src/view/surface.cpp

```diff
@@ -164,7 +164,7 @@ void wf::surface_interface_t::impl::scale_opaque_region(
      * different scales, we just shrink by 1 to compensate for the ceil/floor
      * discrepancy */
     int ceil_factor = 0;
-    if (output->handle->scale != (float)wsurface->current.scale)
+    if (wsurface && output->handle->scale != (float)wsurface->current.scale)
```

Comment:
In the `!wsurface` case we should check whether the output's scale is an integer. If not, we need to expand as well.
So maybe add an `else if (!wsurface && output->handle->scale != (float)(int)output->handle->scale)`

Reply:
Did you mean to double cast output->handle->scale? What would be in the body of this conditional?

Reply:
the same as in the `if` case, which probably means it should be an `||` not else